prov/util/src/util_attr.c

```diff
@@ -1162,8 +1173,7 @@ static uint64_t ofi_get_info_caps(const struct fi_info *prov_info,
 	     (prov_mode & OFI_MR_MODE_RMA_TARGET)))
 		return caps;
 
-trim_caps:
-	return caps & ~(FI_REMOTE_WRITE | FI_REMOTE_READ);
+	return caps;
```

Comment:
Unless I'm reading this function wrong, we call ofi_get_caps() and then return that value in all cases.  So, we don't really need this function at all anymore.  @a-ilango - why were the caps being trimmed, and is it still needed given the other proposed changes?

Reply:
We trim the caps here to prevent providers that require RMA target MR modes from advertising support for RMA target caps when the user doesn't support the required modes. It is still needed. The provider should set correct mr_modes for this function to work correctly.

Does shm provider require any of OFI_MR_MODE_RMA_TARGET modes? If not this function would return the caps without trimming.